subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/DefaultDependencyManagementServices.java

```diff
@@ -242,8 +243,8 @@ DependencyConstraintHandler createDependencyConstraintHandler(Instantiator insta
             return instantiator.newInstance(DefaultDependencyConstraintHandler.class, configurationContainer, dependencyFactory);
         }
 
-        DefaultComponentMetadataHandler createComponentMetadataHandler(Instantiator instantiator, ImmutableModuleIdentifierFactory moduleIdentifierFactory) {
-            return instantiator.newInstance(DefaultComponentMetadataHandler.class, instantiator, moduleIdentifierFactory);
+        DefaultComponentMetadataHandler createComponentMetadataHandler(Instantiator instantiator, ImmutableModuleIdentifierFactory moduleIdentifierFactory, SimpleMapInterner interner) {
+            return instantiator.newInstance(DefaultComponentMetadataHandler.class, instantiator, moduleIdentifierFactory, interner);
```

Comment:
I really think we should collapse these services into one, as noted on the other PR. The constructors are getting ridiculous ;)

Reply:
Yes, I also think it would be better to do something like that, but not in this PR: this would be significant rework, as if we do this, I'd also like to have something that is better fitted to the real DM infrastructure, which plays with modules/selectors/component identifiers and converts one to the other pretty often. There's a lot of room for improvement there.

Reply:
Yes, not blocking this PR, just something I really think we should do to make future reviews simpler.